### DISCUSSION TITLE
Is there a way to configure elsa designer in Project 1 and make the workflows execute in Project 2? all within same solution

### DISCUSSION THREAD

**Discussion by HadiDghaily** [elsa-workflows/elsa-core#3876] [Q&A] [answered]
i have got a solution with multiple projects, lets suppose elsa is configured in project 1, and when set as a startup project in VS, launching it opens the visual designer...etc
i want all workflows set there to be triggered and executed all at project 1, all logic execution to be there.
is this possible? i am using elsa 2.11.0 since its the one on nuget package manager

**Comment by HadiDghaily** [comment]
bump

**Comment by sfmskywalker** [comment]
Yes, the designer is a SPA that you can host from Project 1, and have it read and write workflow definitions  from and to Project 2, from where the workflows will execute.

It's also possible to have Project 1 both host the designer AND store the workflows, but that storage would have to be shared with Project 2, so that it knows about the workflows available to it.

Everything is highly configurable & programmable, so there are other strategies you can employ, but it depends on what kind of decoupling you're after.

**Reply by HadiDghaily** [reply]
"It's also possible to have Project 1 both host the designer AND store the workflows, but that storage would have to be shared with Project 2, so that it knows about the workflows available to it."
Hello! thanks for taking the time to reply to my inquiry
That is exactly what i need, both projects share the same postgres database, project 2 has both the designer and the posrgres persistence.
I would like all workflow configuration and custom activity coding...etc to be done on project 2, but all workflow executions and trigger listening to be done automatically in project1
what is the recommended approach for this?

**Reply by sfmskywalker** [reply]
I probably misunderstood, but it sounds like you want custom activities to be part of Project 2, but workflows to be executed in Project 1?

**Reply by HadiDghaily** [reply]
i want all custom activity definitions, and the management of the workflows to be done in project 2, and all workflows triggered and executed at Project 1

**Reply by sfmskywalker** [reply]
When you say “all workflows triggered and executed in project 1”, can you explain in technical details what that means? For example, if your custom activity and your workflow server is done in project 2, then what does it mean for a workflow to execute in project 1? Do you mean that project 1 should *initiate* workflow execution, but that execution actually happens in project 2?

**Reply by HadiDghaily** [reply]
what im trying to do is seperate all workflow logic and UI into a separate module (project 2), but the place i want the workflows to be effective in is project 1

**Reply by sfmskywalker** [reply]
So when you say Project 2, it’s not necessarily an application, but a class library?

**Reply by HadiDghaily** [reply]
that is right

**Reply by sfmskywalker** [reply]
Then yes, that's a great way to setup your solution.

**Reply by HadiDghaily** [reply]
what would the right methodology for this be? how should elsa be installed and configured into the class library

**Reply by sfmskywalker** [reply]
The class library could expose an extension called 'SetupElsa` method off `IServiceCollection`, which you then invoke from your application's Program/Startup class.

**Reply by HadiDghaily** [reply]
perfect, what would i need for the host app to automatically execute the triggers in the persistence database? and what would the ideal way to host the designer be?

**Reply by sfmskywalker** [reply, marked as answer]
For automatically executing the various triggers, it works the same way for any application hosting the workflow runtime. For more details on setting that up, please check out this documentation: https://v3.elsaworkflows.io/docs/installation/aspnet-apps-workflow-server

Ideally, the designer is hosted in a separate project, for which you can find instructions here: https://v3.elsaworkflows.io/docs/installation/aspnet-apps-workflow-designer

Because the packages aren't released yet to NuGet, you'll need to consume the preview feed. Instructions for that can be found here: https://v3.elsaworkflows.io/docs/installation/packages
